data/scripts/dependency-resolver.py

```diff
@@ -191,16 +191,33 @@ def handle_ccode_check(args, conf, context):
         context.add_kconfig("HAVE_%s" % dep, "bool", "n")
 
 def handle_exec_check(args, conf, context):
-    dep = conf["dependency"].upper()
-    path = which(conf["exec"]) or None
+    dep = conf.get("dependency")
+    exe = conf.get("exec")
+
+    if not exe:
+        print("Could not parse dependency: %s, no exec was specified.")
+        exit(1)
+
+    path = which(exe)
+    required = conf.get("required")
+
+    if required and not path:
+        req_label = context.find_makefile_var("NOT_FOUND")
+        req_label += "executable: %s\\n" % exe
+        context.add_append_makefile_var("NOT_FOUND", req_label, True)
 
-    context.add_cond_makefile_var(dep, path)
+    context.add_cond_makefile_var(dep.upper(), path)
```

Comment:
I mean: inside `add_cond_makefile_var()`, is there any possibility of the first parameter being used being something written in lower case? If there isn't, you can always convert the first parameter to upper case and avoid the trouble of calling `.upper()` everywhere else it's called.